### PR TITLE
Add sqlx example with migrations

### PR DESCRIPTION
**Pull request by FlakM**
Simple use case for sqlx based on tokio postgres example.
Sqlite database is created on execution in ./target directory
and migrations are then run against it.

Also I would like to thank you for creating such a wonderful library! Many kudos for awesome work!

## Motivation

I've got the chance to work with axum (<3!) and put this simple example for showcasing that using sqlx library is very straightforward. Example shows how to setup migrations and automatically create sqlite3 database if none exist at given path.
This example can be trivially used for example to add local backend for `async_cookies` used in sessions example.

## Solution

Add simple example of database creation, migrations execution and connection obtaining via extensions model.

### CODE REVIEW

**Comment by davidpdrsn** [comment]
Thanks!

However we already have an example using [tokio-postgres](https://github.com/tokio-rs/axum/blob/main/examples/tokio-postgres/src/main.rs) which is almost identical to this. So not sure this example adds much that isn't covered by other examples.

**Comment by jplatte** [comment]
> So not sure this example adds much that isn't covered by other examples.

SQLx itself has horrible CI and the maintainers are still kind of slow to merge PRs, yet it is IMO the best SQL library for Rust. I think that for the overall ecosystem, having an example for axum + SQLx in an easy to discover location will be of great value 😄

File: examples/sqlx-sqlite/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        .serve(app.into_make_service())
+        .await
+        .unwrap();
+}
+
+// we can exact the connection pool with `Extension`
+async fn using_connection_pool_extractor(
+    Extension(pool): Extension<SqlitePool>,
+) -> Result<String, (StatusCode, String)> {
+    let (result,) = sqlx::query_as("select username from users where id = $1")
```
Comment (jplatte) [inline examples/sqlx-sqlite/src/main.rs:68]:
```suggestion
    let result = sqlx::query_scalar("select username from users where id = $1")
```

File: examples/sqlx-sqlite/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+        Ok(Self(conn))
+    }
+}
+
+async fn using_connection_extractor(
+    DatabaseConnection(conn): DatabaseConnection,
+) -> Result<String, (StatusCode, String)> {
+    let mut conn = conn;
+    let row: (String,) = sqlx::query_as("select username from users where id = $1")
```
Comment (jplatte) [inline examples/sqlx-sqlite/src/main.rs:103]:
```suggestion
    let row: String = sqlx::query_scalar("select username from users where id = $1")
```

**Review by jplatte — CHANGES_REQUESTED** [review]

**Comment by davidpdrsn** [comment]
Yeah that's true.

In that case I'm thinking if it makes sense to cut the migration stuff, as that's not axum specific. Doing a `select 1+1` should get the point across.

I would also prefer if the example used Postgres but that's less important I guess.

**Comment by FlakM** [comment]
@davidpdrsn I thought that having runnable example without runtime dependencies would be more valuable, especially since there already is tokio-postgres example. I could include some docker snippet for setting pg up though and I will remove migrations if you prefer it this way :+1:

File: examples/sqlx-sqlite/src/main.rs (hunk 2 above)
Comment (FlakM) [reply examples/sqlx-sqlite/src/main.rs:103]: oh thats nicer, thanks!

**Comment by davidpdrsn** [comment]
I think connecting to the default postgres database similarly to how its done [here](https://github.com/tokio-rs/axum/blob/main/examples/tokio-postgres/src/main.rs#L29) would be best. Then there is one less thing for us to maintain/update.

I don't want to get docker involved in the examples. Thats way too much complexity.

**Comment by jplatte** [comment]
What's the problem with the migrations? I've seen a bunch of people try to create their DB via `query` recently, clearly there is a need to show how it's actually done in practice.

**Comment by FlakM** [comment]
You guys work fast :rocket: Give me a moment i will suggest something ;)

**Comment by davidpdrsn** [comment]
> What's the problem with the migrations? I've seen a bunch of people try to create their DB via `query` recently¹, clearly there is a need to show how it's actually done in practice.

I don't think its our job to document how to handle migrations. I would rather users research the available options and pick the one that best suits their app.

**Comment by jplatte** [comment]
But it's an example that shows how to use SQLx.. I know it's possible to use different crates to handle migrations together with SQLx, but using its builtin mirgations is the obvious way to go about it when using it for querying too.

**Comment by davidpdrsn** [comment]
Its an example showing how to integrate sqlx with axum. I would rather keep the examples free to things like this because it forces the user to think about it, instead of just assume that our examples show best practices, which they don't.

I think a more complete example with migrations, docker, etc is more appropriate to link from [ECOSYSTEM.md](https://github.com/tokio-rs/axum/blob/main/ECOSYSTEM.md).

**Comment by FlakM** [comment]
I've removed unwanted bits. If you change you mind I could leave some one off scripts to setup docker or add some functionality

File: examples/sqlx-sqlite/src/main.rs (hunk 1 above)
Comment (FlakM) [reply examples/sqlx-sqlite/src/main.rs]: thanks thats nicer!

**Review by davidpdrsn — APPROVED** [review]
Thanks!

**Comment by FlakM** [comment]
Commit massages and comments always lie!

**Pull request merged**